bmlt-meeting-list.php

```diff
@@ -26,36 +26,71 @@ class Bread {
 		var $formats_all = '';
 		
 		var $service_meeting_result ='';
-		var $optionsName = 'bmlt_meeting_list_options';
+		const SETTINGS = 'bmlt_meeting_list_settings';
+		const OPTIONS_NAME = 'bmlt_meeting_list_options';
+		var $optionsName = Bread::OPTIONS_NAME;
 		var $options = array();
+		var $outside_meeting_result = array();
+		var $allSettings = array();
+		var $maxSetting = 1;
+		var $loaded_setting = 1;
+		
+		function loadAllSettings() {
+		    $this->allSettings = get_option( Bread::SETTINGS );
+		    if ($this->allSettings === false) {
+		        $this->allSettings = array();
+		        $this->allSettings[1] = "Default Setting";
+		        $this->maxSetting = 1;
+		    } else {
+		        foreach ($this->allSettings as $key => $value ) {
+		            if ($key > $this->maxSetting) {
+		                $this->maxSetting = $key;
+		            }
+		        }
+		    }
+		}
+		function startsWith($haystack, $needle)
+		{
+		    $length = strlen($needle);
+		    return (substr($haystack, 0, $length) === $needle);
+		}
 		function __construct() {
 		    $this->protocol = (strpos(strtolower(home_url()), "https") !== false ? "https" : "http") . "://";
-		    if (is_admin() || (isset($_GET['current-meeting-list']) && intval($_GET['current-meeting-list']) == 1)) {
-                $this->getMLOptions();
-                $this->lang = $this->get_bmlt_server_lang();
-
-                if (is_admin()) {
-                    // Back end
-                    //Initialize the options
-                    add_action("admin_init", array(&$this, 'my_sideload_image'));
-                    add_action("admin_menu", array(&$this, "admin_menu_link"));
-                    add_filter('tiny_mce_before_init', array(&$this, 'tiny_tweaks' ));
-                    add_filter('mce_external_plugins', array(&$this, 'my_custom_plugins'));
-                    add_filter('mce_buttons', array(&$this, 'my_register_mce_button'));
-                    add_action("admin_notices", array(&$this, "is_root_server_missing"));
-                    add_action("admin_init", array(&$this, "pwsix_process_settings_export"));
-                    add_action("admin_init", array(&$this, "pwsix_process_settings_import"));
-                    add_action("admin_init", array(&$this, "pwsix_process_default_settings"));
-                    add_action("admin_init", array(&$this, "my_theme_add_editor_styles"));
-                    add_action("admin_enqueue_scripts", array(&$this, "enqueue_backend_files"));
-                    add_action("wp_default_editor", array(&$this, "ml_default_editor"));
-                    add_filter('tiny_mce_version', array( __CLASS__, 'force_mce_refresh' ) );
-                } else if ( intval($_GET['current-meeting-list']) == 1 ) {
-                    $this->bmlt_meeting_list();
-                }
+			if (!($this->startsWith(strtolower($_SERVER['QUERY_STRING']),'current-meeting-list'))&&!is_admin() ) {
```

Comment:
what's the purpose of this line?  could this not be moved to the condition below?

Reply:
Its just the negation of the test that was already there.  We are being called all the time and need to see if we're relevant.   I'm using this negated form because getting the current setting is more complicated (since it might be in a POST), so it makes the code slightly cleaner.

Reply:
what I mean is you can accomplish the same thing by doing `isset($_GET['current-meeting-list'])`.  If this isn't set then you can abandon.